packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -228,6 +228,7 @@ type BaseInputCommonProps = FormInputLabelProps &
      * true if popup is in expanded state
      */
     isPopupExpanded?: boolean;
+    setInputWrapperRef?: (node: HTMLDivElement) => void;
```

Comment:
we already have ref exposed right? what is the ref about?

Reply:
<img width="949" alt="image" src="https://github.com/razorpay/blade/assets/30949385/f45dedfc-f405-4e00-b645-7d0ddd9eacfd">

We have ref exposed for the actual input field (red one in this screenshot)

But to place dropdown and size dropdown correctly with input, we need reference to the wrapper element (blue one in screenshot)

Earlier we had statically defined size of trailing, then checking width of trailing, if trailing exists or not, etc then add that to input element to get final width but thats too complex and likely to break in future. And with floating-ui, it requires even more hacky code to work.